public/js/Histogram.js

```diff
@@ -12,21 +16,28 @@ _.mixin(require('lodash-deep'));
  * @param {array} featureNames - An array of the features used in this screen.
  */
 function Histogram(sampleData, element, featureNames) {
+    var self = this;
```

Comment:
Creating a reference to the ``this`` object is useful for when I went to reference to that particular ``this`` when I'm not in it's scope.

Here's an example from ``lineplot.js``
````
var line = d3.svg.line()
    .x(function(d) { return self.xScale(d[0]); })
    .y(function(d) { return self.yScale(d[1]); });
```

When I call the on method, if I used the ``this`` keyword instead of the ``self` variable I've created, ``this`` will point to the anonymous function rather than the plot object so this won't work.

There is a ``bind`` method, so the above code segment could be written like so:
````
var line = d3.svg.line()
    .x(function(d) { return self.xScale(d[0]); }.bind(this))
    .y(function(d) { return self.yScale(d[1]); }.bind(this));
```
But I personally find it hideous. And there's also situations where you want to call the ``this`` of both the closure you're in, *and* the ``this`` of the other enclosures. See this below example:

````
self.svg = d3.select(this.element).append('svg')
    .attr('width', self.fullWidth)
    .attr('height', self.fullHeight)
    .append('g')
    .attr('transform', 'translate(' + self.margin.left + ',' + self.margin.top + ')')
    .on('click', function() {
        self.onClickUpdate(d3.mouse(this));
    });
````
When handling the click event, I want to call the method ``onClickUpdate`` the belongs to the outer scope, *and* call ``d3.mouse(this)``. If used bind, the wrong reference would get passed to get ``d3.mouse``. 

Open to any suggestions! Like everything in JS, there's no convention. :P